app/models/foreman_salt/concerns/hostgroup_extensions.rb

```diff
@@ -28,31 +28,29 @@ def inherited_salt_module_ids
       end
 
       def salt_proxy
-        return super unless ancestry.present?
-        SmartProxy.find_by_id(inherited_salt_proxy_id)
-      end
-
-      def inherited_salt_proxy_id
         if ancestry.present?
-          self[:salt_proxy_id] || self.class.sort_by_ancestry(ancestors.where('salt_proxy_id is not NULL')).last.try(:salt_proxy_id)
+          SmartProxy.with_features('Salt').find_by_id(inherited_salt_proxy_id)
         else
-          salt_proxy_id
+          super
```

Comment:
What does the super do? Given you already search exactly for the proxy with the right ID, this feels redundant.

Reply:
same behavior is in foreman_chef. Therefore I would stick with this for now.